testsuite/features/support/commonlib.rb

```diff
@@ -751,3 +751,53 @@ def api_unlock
     file.flock(File::LOCK_UN)
   end
 end
+
+# Function to fetch the current list of events
+#
+# @param hostname String The hostname of the system from requested
+def fetch_event_history(hostname)
+  output, _code = get_target('server').run("spacecmd -u admin -p admin system_listeventhistory #{hostname}", check_errors: true)
+  events = []
+  current_event = {}
+
+  output.split("\n").each do |line|
+    case line
+    when /^Id:\s+(\d+)$/
+      current_event[:id] = Regexp.last_match(1).to_i
+    when /^History type:\s+(.+)$/
+      current_event[:history_type] = Regexp.last_match(1)
+    when /^Status:\s+(.+)$/
+      current_event[:status] = Regexp.last_match(1)
+    when /^Completed:\s+(.+)$/
+      begin
+        current_event[:completed] = Time.parse(Regexp.last_match(1))
+      rescue ArgumentError
+        current_event[:completed] = nil
+      end
+    end
+
+    if current_event.key?(:id) && current_event.key?(:completed)
+      events << current_event
+      current_event = {}
+    end
+  end
+
+  events
+end
+
+# Function to get the highest event ID (latest event)
+#
+# @param hostname String The hostname of the system from requested
+def get_last_event_id(hostname)
+  events = fetch_event_history(hostname)
+  last_event = events.max_by { |event| event[:id] }
+  last_event ? last_event[:id] : nil
+end
```

Comment:
We already have implemented a quite similar code, but using API instead of spacecmd.
Please take a look into `features/support/system_monitoring.rb` on `last_onboarding_duration`

```
    node = get_target(host)
    system_id = get_system_id(node)
    events = $api_test.system.get_event_history(system_id, 0, 10)
    onboarding_events = events.select { |event| event['summary'].include? 'certs, channels, packages' }
    last_event_id = onboarding_events.last['id']
    event_details = $api_test.system.get_event_details(system_id, last_event_id)
    # Convert XMLRPC::DateTime to Ruby's Time if necessary
    completed_time = event_details['completed'].is_a?(XMLRPC::DateTime) ? event_details['completed'].to_time : Time.parse(event_details['completed'])
    picked_up_time = event_details['picked_up'].is_a?(XMLRPC::DateTime) ? event_details['picked_up'].to_time : Time.parse(event_details['picked_up'])
```

I suggest to re-use part of this code, so instead of using spacecmd we use the API.